pkg/catalog/manager/traverse.go

```diff
@@ -54,36 +62,42 @@ func (m *Manager) traverseAndUpdate(helm *helmlib.Helm, commit string, cmt *Cata
 		}
 	}
 
-	entriesWithErrors, entriesToProcess := m.preprocessCatalog(catalog, index.IndexFile.Entries)
-	for chart, metadata := range entriesToProcess {
-		newHelmVersionCommits[chart] = v32.VersionCommits{
-			Value: map[string]string{},
+	catalogTemplateMap, err := m.getTemplateMap(catalogName, templateNamespace)
+	if err != nil {
+		return err
+	}
+
+	// Check if a version has been deleted from a template or if an existing template version has changed.
+	hasChartChanged := func(template *v3.CatalogTemplate, metadata helmlib.ChartVersions) bool {
+		if template == nil && metadata == nil {
+			return false
```

Comment:
Hmm should this return true as we discussed? If template is missing(which means template is nil or metadata == nil) then return true. Also you will likely get a panic if template is nil but metadata is not nil. I think this should be `or` condition?

Reply:
I am thinking that we should the logic should be...
If template == nil:
Then means a new template has been added. Metadata is not likely to be nil I think. but if it is nil we can probably return false because a chart has been deleted from upstream then we can skip it and do nothing.

Reply:
Spoke offline and agree on this approach. `metadata` should never be `nil`, so an additional "safety" check has been added to the parent function.